tests/nix-profile.sh

```diff
@@ -140,6 +140,32 @@ printf World2 > $flake2Dir/who
 
 nix profile install $flake1Dir
 [[ $($TEST_HOME/.nix-profile/bin/hello) = "Hello World" ]]
+expect 1 nix profile install $flake2Dir
+diff -u <(nix profile install $flake2Dir 2>&1 1> /dev/null || true) <(cat << EOF
+error: An existing package already provides the following file to the user's profile:
+
+         /tmp/nix-shell.dvSd2u/nix-test/tests/nix-profile/store/d4nwfvm9sz012b4i2331i2rrr5x4byfr-profile-test-4.0/bin/hello
```

Comment:
You'll need to detail with the store dir changing (use `$NIX_STORE_DIR`) and the hash change (not so sure).

Reply:
Yes, the hash is not reproducible on different machines with different test roots. I'm looking for a way to find the store path. I had `nix eval ${flake1Dir}"#default.outPath" --raw`, but it doesn't work yet. I'm going look into this at a later time.

Reply:
Ended up using `nix build --no-link --print-out-paths ${flake2Dir}"#default.out"`.